lib/fizzy/execute.cpp

```diff
@@ -196,6 +196,50 @@ Instance instantiate(const Module& module)
     return instance;
 }
 
+namespace
+{
+template <typename DstT, typename SrcT>
+DstT extend(SrcT in)
+{
+    if constexpr (std::is_signed<SrcT>::value)
+    {
+        using SignedDstT = typename std::make_signed<DstT>::type;
+        return static_cast<DstT>(SignedDstT{in});
+    }
+    else
+        return DstT{in};
+}
+
+template <typename DstT, typename SrcT = DstT>
+inline bool load_from_memory(Instance& instance, Stack<uint64_t>& stack, const uint8_t* immediates)
```

Comment:
`Instance` can be passed by const reference, I guess?

I think I would actually pass only memory here instead of the whole instance.